dapps/src/endpoint.rs

```diff
@@ -39,7 +40,7 @@ pub struct EndpointInfo {
 	pub icon_url: String,
 }
 
-pub type Endpoints = Arc<BTreeMap<String, Box<Endpoint>>>;
+pub type Endpoints = Arc<RwLock<BTreeMap<String, Box<Endpoint>>>>;
```

Comment:
why add `RwLock` if it's only read through? doesn't seem to need any lock at all